lib/random.gi

```diff
@@ -98,11 +96,11 @@ InstallMethod(Reset, [IsGlobalRandomSource, IsObject], function(rs, seed)
   return old;
 end);
 
-InstallMethod(Random, [IsGlobalRandomSource, IsList], function(rs, l)
-  if Length(l) < 2^28 then
-    return RANDOM_LIST(l);
+InstallMethod(Random, [IsGlobalRandomSource, IsInt, IsInt], function(rs, l, u)
+  if u - l + 1 < 2^28 then
+    return RANDOM_LIST([l..u]);
   else
-    return l[Random(rs, 1, Length(l))];
+    return RANDOM_LARGE_RANGE(x -> RANDOM_LIST([0..x]), l, u);
```

Comment:
If it wasn't for the fact that the `io` package already implements a custom random source, I'd suggest to switch the primitive Random to a function `Random(rs, max)` which returns values from the interval `[0..max]` (or `[1..max]` if it was more convenient). Anyway, changing this and syncing it with `io` would be a lot of hassle, so let's leave it at this.

I do wonder what happens if l > u, though... we'll take a random element from an empty list then?

Reply:
(again: these are old issues, I don't expect them to be addressed by this PR)

Reply:
The interface with the two integer arguments is used in many places. I would not change that at this stage.

Reply:
Agreed.